docs/docs/development/setup/setup.md

```diff
@@ -0,0 +1,19 @@
+---
+title: General Info
+sidebar_label: General Information
+---
+
+## Environment Setup
+
+There are 2 ways to setup the zmk development environment: Docker+VSCode and Native OS. The Docker+VSCode approach is a self-contained development environment while Native OS approach will setup your local operating system for development.
+
+The Docker+VSCode approach is very straight forward to setup. The Native approach requires more up-front work but doesn't require installing Docker/VSCode and can be faster for development depending on your computer's hardware.
```

Comment:
It's primarily any think disk IO wise that's slower which is why I left it open-ended. npm in particular can be very sluggish during the ci and run operations during documentation tests.

Also, if the CPU is an Intel atom (doubtful but given folks are using rapsberry pis...) it will probably be slow all around.